src/common/libsubprocess/test/subprocess.c

```diff
@@ -576,6 +577,88 @@ void test_kill_setpgrp (flux_reactor_t *r)
     flux_cmd_destroy (cmd);
 }
 
+void kill_on_exit (flux_subprocess_t *p, flux_subprocess_state_t state)
+{
+    if (state == FLUX_SUBPROCESS_EXITED) {
+        flux_future_t *f = flux_subprocess_kill (p, SIGTERM);
+        /* Note: in local subprocess case, future is returned from
+         * flux_subprocess_kill() fulfilled.
+         */
+        ok (f && flux_future_get (f, NULL) == 0,
+            "flux_subprocess_kill() works after parent exited: (%d) %s",
+            errno,
+            strerror (errno));
+        flux_future_destroy (f);
+    }
+}
+
+void completion_parent_normal_exit (flux_subprocess_t *p)
+{
+    ok (flux_subprocess_state (p) == FLUX_SUBPROCESS_EXITED,
+        "subprocess state == EXITED in completion handler");
+    ok (flux_subprocess_status (p) != -1,
+        "subprocess status is valid");
+    ok (flux_subprocess_exit_code (p) == 0,
+        "subprocess terminated normally");
+    flux_reactor_stop (flux_subprocess_get_reactor (p));
+    completion_sigterm_cb_count++;
+
+    errno = 0;
+    ok (flux_subprocess_kill (p, SIGTERM) == NULL && errno == ESRCH,
+        "flux_subprocess_kill fails with ESRCH");
+
+}
+void test_kill_setpgrp_parent_exited (flux_reactor_t *r)
+{
+    char *av[]  = { TEST_SUBPROCESS_DIR "test_fork_sleep", "100", NULL };
+    flux_cmd_t *cmd = NULL;
+    flux_subprocess_t *p = NULL;
+    int ret;
+
+    ok ((cmd = flux_cmd_create (2, av, environ)) != NULL,
+        "flux_cmd_create");
+    ok (flux_cmd_setenvf (cmd, 1, "TEST_FORK_SLEEP_NOWAIT", "t") == 0,
+        "setenv TEST_FORK_SLEEP_NOWAIT=t");
+
+    flux_subprocess_ops_t ops = {
+        .on_completion = completion_parent_normal_exit,
+        .on_stdout = output_processes_cb,
```

Comment:
if I'm looking at code correctly, `output_processes_cb()` will send a sigterm as well, which I don't think is what you want here.  You probably just want the parent to exit within `test_fork_sleep` and then have the state handler sigterm when that parent exits.

So perhaps the run of the mill output callback should be used instead of `output_processes_cb()`

Reply:
Good point. I did try that and was having trouble, but let me do that again.

Note this branch is causing the job exec "dummy" tests to fail and I haven't been able to determine why yet :shrug:

Reply:
Oh, I forgot. This test sets a `nokill` aux item on the subprocess which prevents `output_processes_cb()` from sending SIGTERM (see above), thereby keeping the test mostly the same as the previous test.

Reply:
You can actually tell it isn't sending SIGTERM because this line under `# kill_setpgrp`
```
ok 130 - flux_subprocess_kill returns future_t
```
Is not printed under `# kill_setpgrp_parent_exited`

Reply:
ahhh sorry I missed that.   Would it be easier to just use one of the other output callbacks in the test file, since you don't actually need `output_processes_cb`?

On the other hand, it is mildly beneficial to have the test look simlar to the other `setpgrp` one, so alternatively maybe just a comment is needed.

Reply:
Yeah, I wanted to make it the same as the existing test.

There's already a comment:
```c
    /* Don't kill subprocess in output handler:
     */
    ok (flux_subprocess_aux_set (p, "nokill", (void *) 0x1, NULL) == 0,
        "flux_subprocess_aux_set (\"nokill\")");
```

Reply:
oh jeez, brain fart :-)   i guess i just got confused, it all makes sense now.  I guess all the bases are covered.